src/components/material/MaterialAvailabilityText/online/helper.ts

```diff
@@ -0,0 +1,42 @@
+import { UseTextFunction } from "../../../../core/utils/text";
+import { ManifestationMaterialType } from "../../../../core/utils/types/material-type";
+
+export type AvailabilityTextMap = Record<
+  string,
+  {
+    text: string;
+    count?: number;
+    limit?: number;
+  }
+>;
+
+export const getAvailabilityText = ({
+  type,
+  map,
+  t
+}: {
+  type: ManifestationMaterialType | "materialIsIncluded";
+  map: AvailabilityTextMap;
+  t: UseTextFunction;
+}) => {
+  if (!map || !map[type]) return null;
+
+  const { text, count, limit } = map[type];
+
+  if (type === "materialIsIncluded") {
+    return t(text);
+  }
+
+  if (count !== undefined && limit !== undefined) {
+    return t(text, {
+      placeholders: {
+        "@count": count,
+        "@limit": limit
+      }
+    });
+  }
+
+  return null;
+};
```

Comment:
Look great. I have come up with a change that is a tad more extensible in the future:
```typescript
export const getAvailabilityText = ({
  type,
  map,
  t,
}: {
  type: ManifestationMaterialType | 'materialIsIncluded';
  map: AvailabilityTextMap;
  t: UseTextFunction;
}) => {
  if (!map || !map[type]) return null;

  const { text, count, limit } = map[type];

  if (count === undefined && limit === undefined && text) {
    return t(text);
  }

  if (count !== undefined && limit !== undefined && text !== undefined) {
    return t(text, {
      placeholders: {
        '@count': count,
        '@limit': limit,
      },
    });
  }

  return null;
};